include/git2/status.h

```diff
@@ -173,12 +173,17 @@ typedef enum {
  * The `pathspec` is an array of path patterns to match (using
  * fnmatch-style matching), or just an array of paths to match exactly if
  * `GIT_STATUS_OPT_DISABLE_PATHSPEC_MATCH` is specified in the flags.
+ *
+ * The `tree` is a tree to be used instead of the head commit's tree for
+ * comparing with the index. For example, this can be useful when preparing
+ * to do an amend commit.
```

Comment:
I don't think that the discussion of an amend commit makes sense in the context of this documentation.  You should also explicitly indicate that it defaults to `HEAD`.   (I know that you're implicitly stating it).  Something like:

> The `tree` is the tree to be used for comparison to the working directory and index; defaults to HEAD.